src/vs/workbench/contrib/terminal/browser/terminalActions.ts

```diff
@@ -128,244 +129,225 @@ export class TerminalLaunchHelpAction extends Action {
-				category,
-				precondition: TerminalContextKeys.processSupported
-			});
-		}
-		async run(accessor: ServicesAccessor, args?: unknown) {
-			const terminalService = accessor.get(ITerminalService);
-			const options = (typeof args === 'object' && args && 'location' in args) ? args as ICreateTerminalOptions : { location: TerminalLocation.Editor };
-			const instance = await terminalService.createTerminal(options);
+	registerTerminalAction({
+		id: TerminalCommandId.CreateTerminalEditor,
+		title: { value: localize('workbench.action.terminal.createTerminalEditor', "Create New Terminal in Editor Area"), original: 'Create New Terminal in Editor Area' },
+		run: async (c, _, args) => {
+			const options = (isObject(args) && 'location' in args) ? args as ICreateTerminalOptions : { location: TerminalLocation.Editor };
+			const instance = await c.service.createTerminal(options);
 			instance.focusWhenReady();
 		}
 	});
 
-	registerAction2(class extends Action2 {
-		constructor() {
-			super({
-				id: TerminalCommandId.CreateTerminalEditorSide,
-				title: { value: localize('workbench.action.terminal.createTerminalEditorSide', "Create New Terminal in Editor Area to the Side"), original: 'Create New Terminal in Editor Area to the Side' },
-				f1: true,
-				category,
-				precondition: TerminalContextKeys.processSupported
-			});
-		}
-		async run(accessor: ServicesAccessor) {
-			const terminalService = accessor.get(ITerminalService);
-			const instance = await terminalService.createTerminal({
+	registerTerminalAction({
+		id: TerminalCommandId.CreateTerminalEditorSide,
+		title: { value: localize('workbench.action.terminal.createTerminalEditorSide', "Create New Terminal in Editor Area to the Side"), original: 'Create New Terminal in Editor Area to the Side' },
+		run: async (c) => {
+			const instance = await c.service.createTerminal({
 				location: { viewColumn: SIDE_GROUP }
 			});
 			instance.focusWhenReady();
 		}
 	});
 
-	registerAction2(class extends Action2 {
-		constructor() {
-			super({
-				id: TerminalCommandId.MoveToEditor,
-				title: terminalStrings.moveToEditor,
-				f1: true,
-				category,
-				precondition: ContextKeyExpr.and(ContextKeyExpr.or(TerminalContextKeys.processSupported, TerminalContextKeys.terminalHasBeenCreated), TerminalContextKeys.terminalEditorActive.toNegated(), TerminalContextKeys.viewShowing)
-			});
-		}
-		async run(accessor: ServicesAccessor) {
-			const terminalService = accessor.get(ITerminalService);
-			terminalService.doWithActiveInstance(instance => terminalService.moveToEditor(instance));
-		}
+	registerActiveInstanceAction({
+		id: TerminalCommandId.MoveToEditor,
+		title: terminalStrings.moveToEditor,
+		precondition: ContextKeyExpr.and(ContextKeyExpr.or(TerminalContextKeys.processSupported, TerminalContextKeys.terminalHasBeenCreated), TerminalContextKeys.terminalEditorActive.toNegated(), TerminalContextKeys.viewShowing),
+		run: (activeInstance, c) => c.service.moveToEditor(activeInstance)
```

Comment:
since there are many `terminalService` functions that do not pass in an instance, could these be aptly named to indicate that they do? 

for instance, `moveInstanceToEditor(activeInstance)`?

that way, the code reads what it actually does - the `terminalService` isn't moving to the editor. the instance is

Reply:
oh, I see that this is called `registerActiveInstanceAction`.... maybe that's good enough?

Reply:
`moveInstanceToEditor` is probably a nicer name for it, bit out of scope of this pr though.